auth/src/main/java/com/firebase/ui/auth/ui/provider/ProviderBase.java

```diff
@@ -1,20 +1,30 @@
 package com.firebase.ui.auth.ui.provider;
 
+import android.arch.lifecycle.LiveData;
+import android.content.Intent;
+import android.support.annotation.LayoutRes;
 import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 import android.support.annotation.RestrictTo;
+import android.support.annotation.StringRes;
 
-import com.firebase.ui.auth.viewmodel.idp.ProviderResponseHandlerBase;
+import com.firebase.ui.auth.IdpResponse;
+import com.firebase.ui.auth.ui.HelperActivityBase;
 
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
-public abstract class ProviderBase implements Provider {
-    private final ProviderResponseHandlerBase mHandler;
+public abstract class ProviderBase {
```

Comment:
Your github comments explaining the class hierarchy were very helpful!  Let's also add class-level JavaDoc comments to these new base/handler classes to explain it to future me when I inevitably confuse myself again.

Reply:
Haha, yep!